### PR TITLE
Added lists

### PR DESCRIPTION
**Pull request by HiQoSolutionsAI**

### CODE REVIEW

File: src/app/board/board.component.html [hunk 1]
```diff
@@ -1 +1,37 @@
...
+            (saveEditableListTitle)="saveEditableListTitle($event)"
+          ></app-list>
+        </mat-card>
+      </ng-container>
+
+      <mat-card class="list-form" [formGroup]="listForm">
+        <mat-card-title>
+          <mat-form-field>
+            <input matInput formControlName="title" placeholder="title" />
+            <mat-error></mat-error>
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.html]: Do you need this empty error tag now?

File: src/app/board/board.component.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+.drag-wrapper {
+  display: flex;
+  flex-wrap: wrap;
+}
+
+.breadcrumbs {
+  display: flex;
+  justify-content: center;
+  margin: 40px 0px 0px 0px;
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.scss]: Can be improved

File: src/app/board/board.component.scss [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  &-button {
+    margin: 0;
+    padding: 15px;
+    text-decoration: none;
+    color: #fff;
+    background-color: #3f51b5;
+  }
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.scss]: Use variables

File: src/app/board/board.component.ts [hunk 4]
```diff
@@ -1,15 +1,84 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable, Subscription } from 'rxjs';
+
+import { BoardsService } from '@app/core/services';
+import { List } from '../core/models/listModel';
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.ts]: Please create and use a barrel import

File: src/app/board/board.component.ts [hunk 5]
```diff
@@ -1,15 +1,84 @@
...
+    return this.boardService.deleteList(listId);
+  }
+
+  saveEditableListTitle({
+    listId,
+    titleValue,
+  }: {
+    listId: string;
+    titleValue: string;
+  }): Promise<void> {
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.ts]: Type can be improved

File: src/app/board/board.component.ts [hunk 6]
```diff
@@ -1,15 +1,84 @@
...
+    };
+
+    await this.boardService.createNewList(list, pushId);
+
+    this.listForm.reset();
+  }
+
+  async getLists(): Promise<void> {
+    this.lists$ = this.boardService.getLists(this.mainBoardId);
+  }
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.ts]: You don't need to use async here

File: src/app/board/board.module.ts [hunk 7]
```diff
@@ -3,10 +3,12 @@ import { CommonModule } from '@angular/common';
...
 import { BoardComponent } from './board.component';
+import { HeaderModule } from '@app/header/header.module';
+import { SharedModule } from '@app/shared/shared.module';
+import { ListComponent } from './list/list.component';
 
 @NgModule({
-  declarations: [BoardComponent],
-  imports: [CommonModule, BoardRoutingModule],
+  declarations: [BoardComponent, ListComponent],
+  imports: [CommonModule, BoardRoutingModule, HeaderModule, SharedModule],
```
Comment (EkaterinaSaverina) [inline src/app/board/board.module.ts:12]: Do we need a HeaderModule here?

File: src/app/core/services/auth.service.ts [hunk 8]
```diff
@@ -4,9 +4,9 @@ import { Store } from '@ngrx/store';
 import firebase from 'firebase/compat/app';
 import { firstValueFrom, Observable } from 'rxjs';
 
-import { User } from '@src/app/core/models/user.model';
-import { FireAuthErrorService } from '@app/core/services';
-import { FirestoreService } from '@app/core/services';
+import { User } from '@app/core/models/user.model';
+import { FireAuthErrorService } from '@app/core/services/';
+import { FirestoreService } from '@app/core/services/';
```
Comment (EkaterinaSaverina) [inline src/app/core/services/auth.service.ts]: Do we need '/' symbol here?

File: src/app/board/board.component.ts [hunk 9]
```diff
@@ -1,15 +1,84 @@
...
+  private initMainBoardsForm(): void {
+    this.listForm = new FormGroup(
+      {
+        title: new FormControl('', [Validators.required]),
+      },
+      { updateOn: 'blur' }
+    );
   }
 
+  ngOnDestroy() {
```
Comment (EkaterinaSaverina) [inline src/app/board/board.component.ts]: Please add return type

File: src/app/dashboard/mainboards/mainboards.component.html [hunk 10]
```diff
@@ -32,10 +37,20 @@ <h4 class="main-board-title" (click)="toggleTitleEditMode()">{{ boardTitle }}</h
...
     (click)="cancelBoardEdit()"
   >
     cancel
   </button>
+
+  <button
+    mat-button
+    class="main-board-link"
+    (click)="getReferredBoardId(boardId)"
+    [ngStyle]="{ display: isTitleEditMode || isDescriptionEditMode ? 'none' : 'block' }"
```
Comment (EkaterinaSaverina) [inline src/app/dashboard/mainboards/mainboards.component.html]: Please correct the attributes order

File: src/app/core/services/boards.service.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { FirestoreService } from '@app/core/services';
+import { Observable } from 'rxjs';
+import { List } from '../models/listModel';
```
Comment (EkaterinaSaverina) [inline src/app/core/services/boards.service.ts]: Please separate imports

File: src/app/dashboard/mainboards/mainboards.component.ts [hunk 12]
```diff
@@ -11,15 +11,16 @@ import {
...
 })
 export class MainBoardsComponent {
   @Input() boardId!: string;
   @Input() boardTitle!: string;
   @Input() boardDescription!: string | undefined;
 
   @Output() deleteBoard = new EventEmitter();
+  @Output() switchToBoard = new EventEmitter();
   @Output() saveEditableBoardTitle = new EventEmitter();
   @Output() saveEditableBoardDescription = new EventEmitter();
```
Comment (EkaterinaSaverina) [inline src/app/dashboard/mainboards/mainboards.component.ts]: Please add types

File: src/app/board/board.module.ts (hunk 7 above)
Comment (HiQoSolutionsAI) [reply src/app/board/board.module.ts:12]: In the next pull request, I will separate the materials module and shared module and then put the header module in shared

**Review by EkaterinaSaverina — APPROVED** [review]

**Pull request merged**